### DISCUSSION TITLE
Change format of Run ID?

### DISCUSSION THREAD

**Discussion by johanneszellinger** [apache/airflow#34064] [Q&A]
Probably a relatively simple (and not the most important) question, but would it be possible to change the format of auto-generated run ids?
For examples, the format for manually triggered run id is `manual__<<Date>>T<<Time>>+ss:ms`. Would it be possible to use a simple `uuid4` instead?

I know one can set the run id when a DAG run is triggered manually, but in my setup I need to generate it within my pipeline from code, currently this is done in a separate python task which precede all other tasks. Using the run id would just be more elegant.

**Comment by Taragolis** [comment]
I'm sure that is not possible to do as default behaviour, just because it will break someone's pipelines/monitoring system.
In the other hand there is another proposal exists for change run_id name generation https://github.com/apache/airflow/discussions/34064, so maybe it is possible to make it extensible, in this case everyone could change it by own implementation if for some reason not satisfied with the default one.

So if you have an idea how it could be achieved I recommend you write ideas to that task/issue.

**Comment by johanneszellinger** [comment]
Hi, thanks for the answer. 
> I'm sure that is not possible to do as default behaviour, just because it will break someone's pipelines/monitoring system.

Yes, I think I was a bit unclear in my formulation. I do not want to make this default behavior, but rather extend/override the function, which generates the run id, same as you describe. 

Your link unfortunately points to this post, but in any case I was just wondering if this is already possible. As mentioned in my particular use case I have already a workaround, just thought this would be neat :)

**Reply by potiuk** [reply]
I think it was about this one: https://github.com/apache/airflow/issues/33945

**Reply by potiuk** [reply]
You can chime in there

**Reply by Taragolis** [reply]
> Your link unfortunately points to this post

🤦 🤦 🤦
